numpy/core/function_base.py

```diff
@@ -87,8 +87,10 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     div = (num - 1) if endpoint else num
 
     # Convert float/complex array scalars to float, gh-3504
-    start = start * 1.
-    stop = stop * 1.
+    # Make sure one can use variables that have an __array_interface__, gh-6634
+    from numpy import asarray
```

Comment:
I think there are some special cases that would break otherwise such as single return value. But there should be tests for that, so go ahead and try ;).